### ISSUE TITLE
/proc/self/stat not accessible on M1 running  x86 read-only filesystem 

### ISSUE THREAD

**Issue by jla** [docker/for-mac#6409]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics

Node process.memoryUsage() calls open("/proc/self/stat", O_RDONLY) on linux.
It fails on M1 running x86 images  with read-only filesystem:
```
Error: EROFS: read-only file system, uv_resident_set_memory
```

### Expected behavior

```
docker run --read-only --rm --platform linux/aarch64  alpine  cat /proc/self/stat                                                                                                                                                               
1 (cat) R 0 1 1 34816 1 4194560 1336 .....
```

### Actual behavior

```
docker run --read-only  --rm --platform linux/amd64  alpine  cat /proc/self/stat
cat: can't open '/proc/self/stat': Read-only file system
```

But it works with relative paths:
```
docker run --read-only  --rm --platform linux/amd64  alpine  cat proc/self/stat                                                                                                                                                    
1 (cat) R 0 1 1 34816 1 4194560 2015 .....
```

**Comment by raimue** [comment]
I can reproduce the same on an amd64 Linux system with Docker 20.10.17 when executing an aarch64 container. The problem is actually in qemu-user-static that is used to execute the binary of the foreign architecture (via binfmt_misc).

1. The syscall `open("/proc/self/stat", ...)` is emulated by qemu by matching the given path
2. For this emulation, qemu wants to [open a temporary file](https://gitlab.com/qemu-project/qemu/-/blob/1f64dd76a1bdf50ec8ab66da0ee7dcfd48ea31e4/linux-user/syscall.c#L8272-8281) in `TMPDIR` (or `/tmp` if the environment variable is not set)
3. If your `/tmp` is not writeable, you see the reported behavior

A possible workaround is to give the Docker container a writable volume with something like `-v /tmp:/tmp`. Here I execute the aarch64 platform variant of the container on my amd64 host system:
```
$ docker run --read-only --rm -it --platform linux/aarch64 -v /tmp:/tmp alpine cat /proc/self/stat
1 (cat) 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 365098343888 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
```

**Comment by raimue** [comment]
Closing the issue is the right action as the issue was with QEMU user emulation. For reference, I submitted a patch to QEMU upstream to use `memfd_create` instead of allocating the file in `/tmp`. The issue is fixed with QEMU 7.1.0. https://github.com/qemu/qemu/commit/5b63de6b54add51822db3c89325c6fc05534a54c
